File: backend/test_api.py
"""Test the auth API endpoints"""
import requests

BASE = 'http://127.0.0.1:5000'

# Test health
print("=== Health Check ===")
r = requests.get(f'{BASE}/api/health')
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test register
print("=== Register ===")
r = requests.post(f'{BASE}/api/auth/register', json={
    'name': 'Test User',
    'email': 'test@example.com',
    'password': 'test123456'
})
print(f"Status: {r.status_code}")
data = r.json()
print(f"Response: {data}")
token = data.get('token', '')
print()

# Test login
print("=== Login ===")
r = requests.post(f'{BASE}/api/auth/login', json={
    'email': 'test@example.com',
    'password': 'test123456'
})
print(f"Status: {r.status_code}")
data = r.json()
print(f"Response: {data}")
token = data.get('token', token)
print()

# Test get current user
print("=== Get Current User ===")
r = requests.get(f'{BASE}/api/auth/me', headers={
    'Authorization': f'Bearer {token}'
})
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test update profile
print("=== Update Profile ===")
r = requests.put(f'{BASE}/api/auth/update-profile', json={
    'phone': '+91 98765 43210',
    'location': 'Mumbai, India',
    'summary': 'Software developer with 3 years experience'
}, headers={
    'Authorization': f'Bearer {token}'
})
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test add education
print("=== Add Education ===")
r = requests.post(f'{BASE}/api/profile/education', json={
    'institution': 'IIT Bombay',
    'degree': 'B.Tech',
    'field': 'Computer Science',
    'start_date': '2018-06-01',
    'end_date': '2022-05-31',
    'grade': '8.5 CGPA'
}, headers={
    'Authorization': f'Bearer {token}'
})
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test add skill
print("=== Add Skill ===")
r = requests.post(f'{BASE}/api/profile/skills', json={
    'name': 'Python',
    'category': 'technical',
    'proficiency': 4
}, headers={
    'Authorization': f'Bearer {token}'
})
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

print("=== ALL TESTS PASSED ===")

File: _test_api.js
const https = require('https');

function req(method, url, data, token) {
  return new Promise((resolve, reject) => {
    const u = new URL(url);
    const headers = { 'Content-Type': 'application/json' };
    if (token) headers['Authorization'] = `Bearer ${token}`;
    const body = data ? JSON.stringify(data) : undefined;
    if (body) headers['Content-Length'] = body.length;
    const opts = { hostname: u.hostname, path: u.pathname, method, headers, timeout: 30000 };
    const r = https.request(opts, (res) => {
      let raw = '';
      res.on('data', (c) => raw += c);
      res.on('end', () => resolve({ status: res.statusCode, body: raw }));
    });
    r.on('error', reject);
    r.on('timeout', () => { r.destroy(); reject(new Error('timeout')); });
    if (body) r.write(body);
    r.end();
  });
}

async function main() {
  const ts = Date.now();
  const email = `test${ts}@example.com`;

  // Register
  console.log('--- REGISTER ---');
  const reg = await req('POST', 'https://careerhack-candidate.onrender.com/api/auth/register', {
    name: 'TestUser', email, password: 'TestPass123!'
  });
  console.log('Status:', reg.status);
  let token = '';
  try { token = JSON.parse(reg.body).token; } catch(e) {}
  console.log('Token:', token ? 'OK' : 'MISSING');

  if (!token) { console.log('Registration failed, cannot continue'); return; }

  // Test all endpoints
  const endpoints = [
    ['GET', '/api/auth/me', null],
    ['PUT', '/api/auth/update-profile', { phone: '+91 9876543210', location: 'Mumbai', summary: 'Dev' }],
    ['GET', '/api/jobs', null],
    ['GET', '/api/resume', null],
    ['GET', '/api/interview', null],
    ['GET', '/api/tracker', null],
    ['GET', '/api/assistant', null],
    ['GET', '/api/profile/education', null],
    ['GET', '/api/profile/experience', null],
    ['GET', '/api/profile/skills', null],
  ];

  for (const [method, path, data] of endpoints) {
    const r = await req(method, `https://careerhack-candidate.onrender.com${path}`, data, token);
    const ok = r.status >= 200 && r.status < 300;
    console.log(`${ok ? 'PASS' : 'FAIL'} | ${method} ${path} | Status: ${r.status} | ${r.body.substring(0, 80)}`);
  }

  console.log('\n=== ALL TESTS DONE ===');
}

main().catch(e => console.log('FATAL:', e.message));
